### DISCUSSION TITLE
how does weblate handle UTF-8 BOMs in general?

### DISCUSSION THREAD

**Discussion by luebbe** [WeblateOrg/weblate#6293] [Q&A] [answered]
Hello, I have the impression that UTF-8 BOM handling isn't consistent throughout weblate.

I found this quite old issue: https://github.com/WeblateOrg/weblate/issues/815, which was resolved by stripping BOMs from uploaded .po files. So I assume the BOMs won't be there when the files are downloaded again.

Then there are the .islu files, where BOMs are preserved. I upload/download them with BOMs intact.
We just tried out uploading .resx files with BOMs, but these lose the BOMs upon download, so I assume they are stripped somewhere on the way, which we are not really happy about.

How does Weblate handle BOMs in general?
How do other users handle BOMs (getting lost in Weblate)?

**Comment by nijel** [comment, marked as answer]
Weblate mostly relies on the translate-toolkit in this and the implementation is not really consistent there

**Reply by luebbe** [reply]
That was the expected (feared ;-)) answer. Thank you!

**Comment by nijel** [comment]
Looking again at #815 only the UTF-8 BOM is stripped in case of merge upload - when the content of file is parsed and used for translation. This has no effect on how the file is stored or downloaded.
